### ISSUE TITLE
Add Specs

### ISSUE THREAD

**Issue by jcraigk** [acook/remedy#16]
This is a useful library. The examples are good. A test suite (RSpec preferred), including unit tests for each class, would be an excellent addition.

**Comment by acook** [comment]
Agreed! As this was originally extracted from a project in 2012 it didn't really come with specs. It would also help explain the details of each capability and use case.
